### ISSUE TITLE
unsubscriptable-object error reported incorrectly

### ISSUE THREAD

**Issue by pbasista** [PyCQA/pylint#2416]
### Steps to reproduce
Run Pylint on the following code:
```python
from typing import Generic, TypeVar

T = TypeVar("T")

class Test(Generic[T]):  # error is reported here
    pass
```

### Current behavior
Pylint reports the following error:
```E1136: Value 'Generic' is unsubscriptable (unsubscriptable-object)```

### Expected behavior
No error should be reported. Python documentation for [`Generic`](https://docs.python.org/3/library/typing.html#typing.Generic) explicitly lists use cases where `Generic` is subscripted.

### pylint --version output
```
$ pylint --version
pylint 2.1.1
astroid 2.0.4
Python 3.7.0 (default, Jun 27 2018, 14:40:03) 
[GCC 8.1.0]
```

This issue is *not* reproducible with Python 3.6. Might be related to issue #2377, which is only reproducible with Python 3.6.

**Comment by pcorpet** [comment]
I believe this is due to Generic using `__class_getitem__`, see [source code](https://github.com/python/cpython/blob/master/Lib/typing.py#L816). And pylint does not handle `__class_getitem__` yet.

**Comment by PCManticore** [comment]
Thanks @pcorpet for the PR!

**Comment by idchlife** [comment]
Thanks for hard work at pylint!

Is it possible to use version with fix for Generic?

**Comment by vyshkant** [comment]
@PCManticore Could you please release this fix?

**Comment by PCManticore** [comment]
@vyshkant I'll do a Pylint release today.

**Comment by belvedere-trading** [comment]
Thank you! I can verify that this fixed the issue on our end.
